### ISSUE TITLE
Supports for React v18+ with Typescript

### ISSUE THREAD

**Issue by CamboimGabriel** [scaleflex/react-cloudimage-responsive#27]
If we try to use it causes an error
<img width="880" alt="image" src="https://user-images.githubusercontent.com/36650520/229200409-6eb93260-549e-4ddb-9be3-5dbf2d028c86.png">

**Comment by unitedustom** [comment]
Yeah i'm getting the same problem when trying to use it with Typescript.

**Comment by davepenfold** [comment]
Hello,
I can offer some suggestions, I have two questions about the type definitions for this project:

1. https://github.com/scaleflex/react-cloudimage-responsive/blob/26be77afcae525de820f0ff148bf23e2d20291de/index.d.ts#L33 Shouldn't `doNotReplaceURL` be an optional property like all the others? If this property is omitted it causes a TS error.
2. https://github.com/scaleflex/react-cloudimage-responsive/blob/26be77afcae525de820f0ff148bf23e2d20291de/index.d.ts#L22 `children` property is not defined here, I believe it should be? This is what is likely causing the TS error provided in this issue. It could be of type `React.ReactNode` ?
